tools/issue-labeler/src/IssueLabelerService/Answers/OpenAiAnswerService.cs

```diff
@@ -110,10 +142,87 @@ public async Task<AnswerOutput> AnswerQuery(IssuePayload issue, Dictionary<strin
 
             _logger.LogInformation($"Open AI Response for {issue.RepositoryName} using the Complete Triage model for issue #{issue.IssueNumber}.: \n{formatted_response}");
 
-            return new AnswerOutput {
-                Answer =  formatted_response, 
-                AnswerType = solution ? "solution" : "suggestion" 
+            return new AnswerOutput
+            {
+                Answer = formatted_response,
+                AnswerType = solution ? "solution" : "suggestion"
+            };
+        }
+
+        private void AddUniqueItems<T, TKey>(
+            IEnumerable<T> items,
+            HashSet<TKey> uniqueSet,
+            List<T> aggregatedList,
+            Func<T, TKey> keySelector)
+        {
+            foreach (var item in items)
+            {
+                if (uniqueSet.Add(keySelector(item)))
+                {
+                    aggregatedList.Add(item);
+                }
+            }
+        }
+
+        private async Task RetrieveAndAggregateSubqueryResults(
+            IEnumerable<string> subqueries,
+            string issueIndexName,
+            string issueSemanticName,
+            string issueFieldName,
+            string documentIndexName,
+            string documentSemanticName,
+            string documentFieldName,
+            int top,
+            double scoreThreshold,
+            Dictionary<string, string> labels,
+            HashSet<Issue> uniqueIssues,
+            HashSet<Document> uniqueDocs
+        )
+        {
+            var subqueryTasks = subqueries.Select(async subquery =>
+            {
+                var subqueryIssues = await _ragService.SearchIssuesAsync(issueIndexName, issueSemanticName, issueFieldName, subquery, top, scoreThreshold, labels);
+                var subqueryDocs = await _ragService.SearchDocumentsAsync(documentIndexName, documentSemanticName, documentFieldName, subquery, top, scoreThreshold, labels);
+                foreach (var subqueryIssue in subqueryIssues)
+                    uniqueIssues.Add(subqueryIssue);
+                foreach (var subqueryDoc in subqueryDocs)
+                    uniqueDocs.Add(subqueryDoc);
+            });
+            await Task.WhenAll(subqueryTasks);
+        }
+
+        private Dictionary<string, string> BuildUserPromptData(
+            List<Issue> allIssues,
+            List<Document> allDocs,
```

Comment:
```suggestion
            IEnumerable<Issue> allIssues,
            IEnumerable<Document> allDocs,
```

We could also use `ICollection<T>` if we think there's a need to mutate, but I don't see us needing that.